repository/documents_api.go

```diff
@@ -1018,6 +1118,24 @@ func (a *DocumentsService) verifyUpdateRequest(
 			"no import directive permission")
 	}
 
+	isMeta := req.UpdateMetaDocument ||
+		(req.Document != nil && isMetaURI(req.Document.Uri))
```

Comment:
Thanks for prodding me about how hard the logic was to follow, made me take another stab at it. I pushed a new commit that separates out most of the behaviour that is conditional on the document being a meta doc. The access check in the end is also presents two clearer variants in handling, rather than the inline variations it had before. Let me know what you think tomorrow!